### DISCUSSION TITLE
HELP: ESP32-WROVER-DEV + SSD1322

### DISCUSSION THREAD

**Discussion by holefrog** [sle118/squeezelite-esp32#353] [Q&A] [answered]
Hi,
I am working on  ESP32-WROVER-DEV + SSD1322.
So far I have two questions:
1. Booting 
When powered up, it won't start. You must press the EN/RST button to boot it up; otherwise, it will remain in 'Terminal ready' status.

UART result:
```
picocom v3.1

port is        : /dev/ttyUSB0
flowcontrol    : none
baudrate is    : 115200
parity is      : none
databits are   : 8
stopbits are   : 1
escape is      : C-a
local echo is  : no
noinit is      : no
noreset is     : no
hangup is      : no
nolock is      : no
send_cmd is    : sz -vv
receive_cmd is : rz -vv -E
imap is        : 
omap is        : 
emap is        : crcrlf,delbs,
logfile is     : none
initstring     : none
exit_after is  : not set
exit is        : no

Type [C-a] [C-h] to see available commands
Terminal ready
```

2. Display
It can correctly display 'Analog VU Meter and Remaining Time.' However, when choosing 'Spectrum Analyzer and Progress Bar and Remaining Time,' there is no Spectrum Analyzer.
![PXL_20231105_221433757 MP](https://github.com/sle118/squeezelite-esp32/assets/17289895/d990a547-32bb-465b-acf3-062b8ac8ee28)
![PXL_20231105_221411227 MP](https://github.com/sle118/squeezelite-esp32/assets/17289895/236bb1e9-514b-48b0-bd1f-3fdf35c8b635)

**Comment by nonzhe** [comment]
How does wrover connect to SD1322? need to connect both Reset and DC pin?

**Reply by holefrog** [reply]
In NVS, "display_config"
```
SPI,width=256,height=64,reset=23,cs=5,speed=8000000,driver=SSD1322
```

In NVS, "spi_config":
```
data=19,clk=18,dc=22
```

**Comment by philippe44** [comment]
How is your configuration of display in the LMS's players settings? Re reset issue, it feels like the UART connection locks the EN pin

**Reply by holefrog** [reply]
Here is LMS
![Screenshot from 2023-11-06 15-50-16](https://github.com/sle118/squeezelite-esp32/assets/17289895/57d0d411-9d16-4b00-82b6-349c5831c080)

**Reply by holefrog** [reply]
"For  reset:  it feels like the UART connection locks the EN pin"
What can I do to fix the problem? I am using Ali-Express ESP32-WROVER-CAM, the pin outs is below:
![ESP32_Pinout](https://github.com/sle118/squeezelite-esp32/assets/17289895/ee7d19b3-5f08-41fa-9221-e45cff409295)

**Reply by philippe44** [reply]
That's likely the UART connection to using DTR or CTS (sorry I don't remember) that might miss a pull up.

**Reply by philippe44** [reply]
Re display, sorry I meant what's in "basic settings"

**Comment by holefrog** [comment]
> Re display, sorry I meant what's in "basic settings"

Sorry to waste your time on this issue.
I do not know what have happened. But now, when I chose 'Spectrum Analyzer and Progress Bar and Remaining Time,' it can display the Spectrum correctly. It might be my browser's issue last time.

**Comment by holefrog** [comment]
> That's likely the UART connection to using DTR or CTS (sorry I don't remember) that might miss a pull up.
Is there any way to solve this problem, such as connecting the EN to VCC?

**Reply by philippe44** [reply]
EN too Vcc via a resistor of 10k min maybe, unless your uart board does a pulldown.

**Comment by holefrog** [comment]
The EN is 3.3V. When connecting EN to GND, it will reset.

It is a little bit strange. When it arrived, it could boot somehow (I can see a loop running from terminal)  when powered on.


Get BlueMail for Android<https://bluemail.me>
On Nov 7, 2023, at 1:50 p.m., philippe44 ***@***.******@***.***>> wrote:

EN too Vcc via a resistor of 10k min maybe, unless your uart board does a pulldown.

**Comment by philippe44** [comment]
So I don't know what happens between that board and your usb-to-uart. Probably a scope would be necessary.

**Comment by holefrog** [comment]
I have a very simple logical analyzer. What should I look for?

Regards,


Get BlueMail for Android<https://bluemail.me>
On Nov 7, 2023, at 3:18 p.m., philippe44 ***@***.******@***.***>> wrote:

So I don't know what happens between that board and your usb-to-uart. Probably a scope would be necessary.

**Reply by philippe44** [reply]
Just try to see what happens on EN when you start and reboot

**Reply by holefrog** [reply]
Here is the EN pin output:
It will keep high after power on.

![Screenshot from 2023-11-07 17-58-46](https://github.com/sle118/squeezelite-esp32/assets/17289895/c52bc4a3-e7c2-47ca-83b4-45d8e139f7f6)

**Comment by philippe44** [comment]
I'm getting a bit confused now, can we summarize the issue? When you power on the board, it would not start, until you reset it. Does it happen regardless of the fact that a UART is connected or not?

**Comment by holefrog** [comment]
Sorry, my bad.

Yes.You correctly descripted the issue:

When power on the board, it would not start, until press reset.  It happens regardless of the fact that a UART is connected or not.



On Nov 7, 2023, at 6:09 p.m., philippe44 ***@***.******@***.***>> wrote:

I'm getting a bit confused now, can we summarize the issue? When you power on the board, it would not start, until you reset. Does it happen regardless of the fact that a UART is connected or not?

**Reply by philippe44** [reply]
ok, so I was going on the wrong track. It's a board problem then

**Reply by philippe44** [reply, marked as answer]
And I should add that usually it's the capacitor on the EN signal that is not large enough (or I should say the RC) which means the EN is released while Vcc is not stable enough and the chip cannot boot. Well-known problem and others (including myself on SqueezeAMP) had the issue.

**Reply by holefrog** [reply]
Hi, philippe44. I apologize for taking up so much of your time and effort with the well-known problem. Following your instructions, I connected a 10uF capacitor to EN and GND. Now, it can boot when powered on. Thank you so much.
